### PR TITLE
feat: implement header

### PR DESCRIPTION
**Pull request by hitendreamworld**
- render title and X(close) button

### CODE REVIEW

File: dw-menu.js [hunk 1]
```diff
@@ -105,10 +145,24 @@ export class DwMenu extends DwCompositeDialog {
 
   connectedCallback() {
     this._setDialogConfig();
+    this._showHeader = Boolean(this.heading) || this.showClose;
```
Comment (ChiragMoradiya) [inline dw-menu.js]: @hitendreamworld computed property value should be set from `willUpdate` in general practice. So, please update this implementation

File: dw-menu.js [hunk 2]
```diff
@@ -20,6 +26,35 @@ export class DwMenu extends DwCompositeDialog {
...
+      :host([type="modal"][_showHeader]) .mdc-dialog__title {
+        height: 56px;
+      }
+
+      :host([type="modal"]) .mdc-dialog__title::before {
+        height: auto;
+      }
+
+      dw-icon-button {
+        width: max-content;
```
Comment (ChiragMoradiya) [inline dw-menu.js]: I don't understand why this css is needed for dw-icon-button

File: dw-menu.js [hunk 3]
```diff
@@ -105,10 +145,24 @@ export class DwMenu extends DwCompositeDialog {
...
     super.connectedCallback();
   }
 
+  get _headerTemplate() {
+    return html`
+      ${this.showClose
+        ? html`<dw-icon-button
+            icon="close"
+            dismiss
+            @click=${(e) => console.log("close: invoked", e)}
```
Comment (ChiragMoradiya) [inline dw-menu.js]: remove this `console.log` if not needed.

**Review by ChiragMoradiya — CHANGES_REQUESTED** [review]

File: dw-menu.js (hunk 1 above)
Comment (hitendreamworld) [reply dw-menu.js]: DwCompositeDialog uses the old `lit-element` version and that library does not provide `willUpdate` lifecycle. So, I wrote a custom getter/setter for `heading` and `showClose`.

**Pull request merged**